### PR TITLE
MIGENG-152 Export to CSV - Workload Inventory Report

### PR DESCRIPTION
**Pull request by carlosthe19916**
Added `/report/{id}/workload-inventory/csv` endpoint which should export all workloadkInventories of one Analysis.

https://jira.coreos.com/browse/MIGENG-152

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 1]
```diff
@@ -54,19 +61,36 @@
...
     private String datacenter;
+
+    @DataField(pos = 3, columnName = "Cluster")
     private String cluster;
 
+    @DataField(pos = 4 , columnName = "VM name")
     @Column(name = VM_NAME)
     private String vmName;
 
+    @DataField(pos = 5, columnName = "OS name")
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    @DataField(pos = 5, columnName = "OS type")
```

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 2]
```diff
@@ -54,19 +61,36 @@
...
     @Column(name = OS_NAME)
     private String osName;
+
+    @DataField(pos = 6, columnName = "OS description")
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    @DataField(pos = 6, columnName = "Operating system description")
```

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 3]
```diff
@@ -75,8 +99,11 @@
     )
     private Set<String> workloads;
 
+    @DataField(pos = 11, columnName = "Complexity")
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    @DataField(pos = 11, columnName = "Effort")
```

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 4]
```diff
@@ -86,13 +113,16 @@
             }
     )
     private Set<String> recommendedTargetsIMS;
+
+    @DataField(pos = 13, columnName = "Flags")
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    @DataField(pos = 13, columnName = "Flags IMS")
```

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 5]
```diff
@@ -54,19 +61,36 @@
...
+    @DataField(pos = 7, precision = 2, columnName = "Disk space")
     private Long diskSpace;
+
+    @DataField(pos = 8, columnName = "Memory")
     private Long memory;
+
+    @DataField(pos = 9, columnName = "CPU cores")
     private Integer cpuCores;
+
+    @DataField(pos = 10, columnName = "Workloads")
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    @DataField(pos = 10, columnName = "Workload")
```

**Review by mrizzi — COMMENTED** [review]
I've added some comments to have the CSV heading consistent with the WI table heading.

**Comment by carlosthe19916** [comment]
Thanks @mrizzi for pointing out the correct name of the CSV file headers

**Review by mrizzi — APPROVED** [review]
The code looks good to me and the test has been successful.
Thanks @carlosthe19916 :+1:

**Pull request merged**